catalog/ui/catalog-ui-search/src/main/webapp/main.js

```diff
@@ -100,6 +101,13 @@ require([
     $(document).ready(function () {
         document.title = properties.branding + ' ' + properties.product;
     });
-    // Actually start up the application.
-    app.App.start({});
+    // Actually start up the application.  Entire app depends on workspaces, so don't allow anything until they're fetched.
+    var workspaces = store.get('workspaces');
+    if (workspaces.fetched){
+        app.App.start({});
+    } else {
+        workspaces.once('sync', function(){
+            app.App.start({});
```

Comment:
In the future we may want to gracefully handle failures to prevent being stuck here forever